aurora/contracts/src/AuroraErc20FastBridge.sol

```diff
@@ -320,36 +326,53 @@ contract AuroraErc20FastBridge is Initializable, UUPSUpgradeable, AccessControlU
         IEvmErc20 token = tokenInfo.auroraTokenAddress;
         require(tokenInfo.isStorageRegistered == true, "The token storage is not registered");
 
-        require(address(token) != address(0), "The token is not registered");
+        require(address(token) != address(0) ||
+                UtilsFastBridge.isStrEqual(transferMessage.transferTokenAccountIdOnNear, auroraEngineAccountIdOnNear),
+                "The token is not registered");
 
         uint256 totalTokenAmount = uint256(transferMessage.transferTokenAmount + transferMessage.feeTokenAmount);
 
-        token.transferFrom(msg.sender, address(this), totalTokenAmount);
+        if (address(token) != address(0)) {
+            require(msg.value == 0, "Incorrect amount of Ether attached");
 
-        // WARNING: The `withdrawToNear` method works asynchronously.
-        // As a result, there is no guarantee that this method will be completed before `initTransfer()`.
-        // In case of such an error, the user will be able to call the `withdraw()` method and get his tokens back.
-        // We expect such an error not to happen as long as transactions are executed in one shard.
-        token.withdrawToNear(bytes(getImplicitNearAccountIdForSelf()), totalTokenAmount);
+            token.transferFrom(msg.sender, address(this), totalTokenAmount);
+
+            // WARNING: The `withdrawToNear` method works asynchronously.
```

Comment:
This comment can be removed due to the this explanation https://github.com/aurora-is-near/aurora-contracts-sdk/pull/13#discussion_r1330092099

Reply:
fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/113/commits/3bcebe661a3a0ccf60e139020d774085a4354940